engine/src/test/java/org/kigalisim/validate/LiveTests.java

```diff
@@ -442,7 +609,81 @@ public void testBasicReplace() throws IOException {
     assertTrue(resultSubB5.getManufacture().getValue().doubleValue() > 0,
         "Sub B should have non-zero manufacture in year 5 due to replacement");
   }
-  */
+
+  /**
+   * Test basic_replace_units.qta produces expected values.
+   * This tests units-based replacement.
+   */
+  @Test
+  public void testBasicReplaceUnits() throws IOException {
+    // Load and parse the QTA file
+    String qtaPath = "../examples/basic_replace_units.qta";
+    ParsedProgram program = KigaliSimFacade.parseAndInterpret(qtaPath);
+    assertNotNull(program, "Program should not be null");
+
+    // Get the scenario
+    String scenarioName = "Sim";
+    ParsedScenario scenario = program.getScenario(scenarioName);
+    assertNotNull(scenario, "Scenario should exist");
+
+    // Get start and end years from scenario
+    int startYear = scenario.getStartYear();
+    int endYear = scenario.getEndYear();
+    assertEquals(1, startYear, "Start year should be 1");
+    assertEquals(10, endYear, "End year should be 10");
+
+    // Create engine and machine
+    Engine engine = new SingleThreadEngine(startYear, endYear);
+    PushDownMachine machine = new SingleThreadPushDownMachine(engine);
+
+    // Execute the default policy
+    ParsedPolicy defaultPolicy = program.getPolicy("default");
+    assertNotNull(defaultPolicy, "Default policy should exist");
+    executePolicy(defaultPolicy, machine);
+
+    // Execute other policies in the scenario
+    for (String policyName : scenario.getPolicies()) {
+      ParsedPolicy policy = program.getPolicy(policyName);
+      assertNotNull(policy, "Policy " + policyName + " should exist");
+      executePolicy(policy, machine);
+    }
+
+    // Verify results for year 1 (before replacement)
+    List<EngineResult> results = engine.getResults();
+
+    // Check Sub A in year 1
+    EngineResult resultSubA1 = findResult(results, "Test", "Sub A", 1);
+    assertNotNull(resultSubA1, "Should have result for Test/Sub A in year 1");
+    assertEquals(100000.0, resultSubA1.getManufacture().getValue().doubleValue(), 0.0001,
+        "Sub A Manufacture should be 100000 kg in year 1");
+
+    // Check Sub B in year 1
+    EngineResult resultSubB1 = findResult(results, "Test", "Sub B", 1);
+    assertNotNull(resultSubB1, "Should have result for Test/Sub B in year 1");
+    assertEquals(0.0, resultSubB1.getManufacture().getValue().doubleValue(), 0.0001,
```

Comment:
These should use `getGhgConsumption` similar to the JS version. We should add `getGhgConsumption` if it is not available in Java.